Assets/Scripts/ssjAug2018/World/Mailbox/MailboxManager.cs

```diff
@@ -109,9 +89,36 @@ public void ActivateMailboxGroup()
                 letterCount = (letterCount > setSize) ? setSize : letterCount;
                 
                 box.ActivateMailbox(letterCount);
+                _activeMailboxes++;
                 validBoxes.Remove(box);
                 setSize =- letterCount;
             }
         }
+
+        private List<Mailbox> GetMailboxesInRange(Transform origin, float minimum, float maximum)
+        {
+            List<Mailbox> validBoxes = new List<Mailbox>();
+            Physics.OverlapSphereNonAlloc(origin.position, maximum, hits, LayerMask.GetMask("Mailboxes"));
+            foreach(Collider box in hits)
```

Comment:
so next bit on this - you can't foreach it. OverlapSphereNonAlloc() returns the number of hits and you'll have to do a normal for loop over that amount in the array. and 50 is probably way too many. I'd start with like 10.